mlir/utils/jenkins/Jenkinsfile

```diff
@@ -236,13 +241,10 @@ pipeline {
                             }
                         }
                         stage("Build MIOpen with libMLIRMIOpen") {
-                            when { allOf {
-                                equals expected: true, actual: params.nightly;
-                                anyOf {
-                                    equals expected: true, actual: params.perfTest;
-                                    equals expected: true, actual: params.canXdlops;
-                                } }
-                            }
+                            when { anyOf {
+                                    equals expected: true, actual: params.nightly;
+                                    equals expected: true, actual: params.weekly;
+                            }}
```

Comment:
Let's keep the control constants in here

Reply:
What do you mean?
With this change, the stage can be run for nightly+staticLib (MIOpen tests), nightly+perfTest, or weekly+staticLib. If canXdlops=true, it is on a private CI node; otherwise, on a public CI node.

Reply:
I mean that we don't have this stage restricted to (perfTest || staticLib) && (nightly || weekly)

Reply:
Changed it to weekly && staticLib.

Reply:
Ok, just to make sure, this test needs to run in every nightly, but only in weeklies where we're doing static library tests?

`nightly || (weekly && staticLib)` doesn't feel right.

And, as a followup, why is this now `staticLib` and not `perfTest`?

Reply:
Ah. In which case, so as to not break the selective CI setup, can you make this `(nightly || weekly) && staticLib`

Reply:
Sorry. I said wrong. This test is run (nightly&&perfTest) || (nightly&&staticLib) || (weekly&&staticLib). The condition (perfTest || staticLIb) has been specified for the top-level stage "Static Library". Since perfTest is only done in nightlies, I disallow it for weeklies.

Reply:
Ah. That makes sense, thanks for clarifying!